src/components/MessagesList/MessagesList.vue

```diff
@@ -366,23 +366,38 @@ export default {
 			const { request, cancel } = CancelableRequest(fetchMessages)
 			// Assign the new cancel function to our data value
 			this.cancelFetchMessages = cancel
+
+			const token = this.token
+			const lastKnownMessageId = this.$store.getters.getFirstKnownMessageId(token)
+			let newestKnownMessageId = 0
+
 			// Make the request
 			try {
-				const token = this.token
-				const lastKnownMessageId = this.$store.getters.getFirstKnownMessageId(token)
 				const messages = await request({ token, lastKnownMessageId, includeLastKnown: includeLastKnown ? '1' : '0' })
 				// Process each messages and adds it to the store
 				messages.data.ocs.data.forEach(message => {
 					if (message.actorType === 'guests') {
 						this.$store.dispatch('setGuestNameIfEmpty', message)
 					}
 					this.$store.dispatch('processMessage', message)
+					newestKnownMessageId = Math.max(newestKnownMessageId, message.id)
 				})
 
 				if (messages.headers['x-chat-last-given']) {
 					this.$store.dispatch('setFirstKnownMessageId', {
 						token: token,
-						id: messages.headers['x-chat-last-given'],
+						id: parseInt(messages.headers['x-chat-last-given'], 10),
+					})
+				}
+
+				// For guests we also need to set the last known message id
+				// after the first grab of the history, otherwise they start loading
+				// the full history with getNewMessages().
+				if (includeLastKnown && newestKnownMessageId
```

Comment:
As the last known message id is only set if it was not already set, is it really needed to limit this to `includeLastKnown` being enabled too?

Reply:
Yes, otherwise a subsequent getOldMessages() call would set it's own newest message as last known, and then we would fetch the messages of the first getOldMessages() call with the next getNewMessages() call.

Reply:
> a subsequent getOldMessages() call would set it's own newest message as last known

But in a subsequent `getOldMessages()` call the last known message id would be already set, so it would not be set again, no? Anyway, this is already merged, so... :-P